src/App.tsx

```diff
@@ -10,21 +11,56 @@ import TableContainer from '@mui/material/TableContainer';
 import TableRow from '@mui/material/TableRow';
 import Paper from '@mui/material/Paper';
 import { TableHead } from "@mui/material";
+import { log_levels } from "./components/Log_Levels.tsx";
 
-type getMessageReturn = [string[],string[],string[],string[],number[],string[]]
+class MessageReturn {
+  timestamp: string[] = [];
+  host: string[] = [];
+  log_level_str: string[] = [];
+  log_message: string[] = [];
+  log_level: number[] = [];
+  app_name: string[] = [];
+}
+
+interface LogMessage {
+  timestamp: string;
+  source: string;
+  application_name: string;
+  level: number;
+  full_message: string;
+}
+
+interface Message {
+  [key: string]: LogMessage;
+}
+
+interface LogData {
+  results: {
+      [key: string]: {
+        search_types: {
+          [key: string]: {
+            messages: Message[];
+          };
+      };
+  };
+}}
```

Comment:
Good! I think these should go in their own source file to simplify things. I like the pato lays it out if you want to base it on something: https://github.com/DiamondLightSource/pato-frontend/blob/master/src/schema/interfaces.ts